src/CreateEntityMappingView/CreateEntityMappingView.js

```diff
@@ -249,98 +312,192 @@ class CreateEntityMappingView extends ConnectionComponent {
+						fullWidth={true}>
+						{config.statuses.map((status, key) => {
+							if (status.name !== "UNDEFINED") {
+								return <MenuItem
+									key={key}
+									value={status.name}
+									primaryText={status.name}/>;
+							}
+							return <MenuItem
+								value={"None"}
+								primaryText={"None"}
+								key={key}/>;
+						})}
+					</SelectField>
+				</Col>
+			</Row>
+		);
+	}
+
+	renderSelectTypes(initialData) {
+		return (
+			<Row>
+				<Col md={6}>
+					<SelectField
+						value={this.state.selectedEventType.value}
+						errorText={this.state.selectedEventType.errorMessage}
+						onChange={this.handleEventTypeChange}
+						floatingLabelText="Select Event Type"
+						fullWidth={true}>
+						{CreateEntityMappingView.getMenuItems(initialData.eventTypes)}
+					</SelectField>
+				</Col>
+				<Col md={6}>
+					<SelectField
+						value={this.state.selectedEntityType.value}
+						errorText={this.state.selectedEntityType.errorMessage}
+						onChange={this.handleEntityTypeChange}
+						floatingLabelText="Select Entity Type"
+						fullWidth={true}>
+						{CreateEntityMappingView.getMenuItems(initialData.entityTypes)}
+					</SelectField>
+				</Col>
+			</Row>
+		);
+	}
+
+	static renderAttributeFields(initialData) {
+		return initialData.attributeFields;
+	}
+
+	static renderButtons(submitCallback) {
+		return (
+			<div className={css(AppStyles.textAlignCenter)}>
+				<RaisedButton
+					label="Abort"
+					icon={<IconCancel/>}
+					className={css(AppStyles.marginAllSites)}
+					secondary={true}
+					onTouchTap={CreateEntityMappingView.abort}
+				/>
+				<RaisedButton
+					label="Save"
+					icon={<IconSave/>}
+					onTouchTap={submitCallback}
+					className={css(AppStyles.marginAllSites)}
+					primary={true}
+				/>
+			</div>
+		);
+	}
+
+	renderComponentBody(initialData, optionalActions, submitCallback) {
+		return (
+			<Container>
+				<div>
+					{optionalActions && optionalActions.rejected && <ErrorMessage message={optionalActions.reason} />}
+					{this.renderSelectTargetStatus()}
+					{this.renderSelectTypes(initialData)}
+					{CreateEntityMappingView.renderAttributeFields(initialData)}
+					{CreateEntityMappingView.renderButtons(submitCallback)}
+				</div>
+			</Container>
+		);
+	}
+
+	fetchInitialData() {
 		const allFetches = PromiseState.all([this.props.eventTypes, this.props.entityTypeHierarchy]);
 		const eventTypes = this.props.eventTypes.value;
-		const optionalActions = this.props.createEntityMappingResponse;
-		if(optionalActions && optionalActions.fulfilled) {
-			window.history.back();
-			return null;
-		}
-		const entityTypes = this.transformHierarchy(this.props.entityTypeHierarchy.value);
+		const entityTypes = CreateEntityMappingView.transformHierarchy(this.props.entityTypeHierarchy.value);
 		const connectionIncomplete = super.render(allFetches);
 		if (connectionIncomplete) {
-			return connectionIncomplete;
+			return false;
```

Comment:
I am not sure you should return false here. 
connectionIncomplete calls the render method of ConnectionComponent (via super) and returns either a Loading animation or a ErrorMessage if the loading procedure has not finished yet.